main/pipeline/pipeline.cc

```diff
@@ -207,8 +228,18 @@ ast::ParsedFile indexOne(const options::Options &opts, core::GlobalState &lgs, c
             if (file.data(lgs).strictLevel == core::StrictLevel::Ignore) {
                 return emptyParsedFile(file);
             }
-            auto parseTree = runParser(lgs, file, print, opts.traceLexer, opts.traceParser);
```

Comment:
This is not the only call to `runParser`—there is another call in `desugarOne`, which is a method used only by the extract variable code action.

The Extract Variable code action has some special logic that makes desguar behave differently, and I wanted to flag that as something else that we'll want to double check whenever the approach shifts from "prism -> parser::Node -> ast::ExpressionPtr" to just "prism -> ast::ExpressionPtr"